finalfrontier/src/vocab.rs

```diff
@@ -388,6 +392,42 @@ where
         SimpleVocab::new(builder.config, types, builder.n_items)
     }
 }
+/// Constructs a `SimpleVocab<S>` from a `VocabBuilder<T>` where `T: Into<S>`.
+impl<T, S> From<VocabBuilder<SimpleSizedVocabConfig, T>> for SimpleVocab<S>
+where
+    T: Hash + Eq + Into<S>,
+    S: Hash + Eq + Clone + Ord,
+{
+    fn from(builder: VocabBuilder<SimpleSizedVocabConfig, T>) -> Self {
+        let config = builder.config;
+
+        let mut types: Vec<_> = builder
```

Comment:
This now also has duplication that can be avoided using the `enum` suggested above.